### DISCUSSION TITLE
Different search spaces

### DISCUSSION THREAD

**Discussion by AbrightWay** [optuna/optuna#5503] [Q&A] [answered]
Dear Optuna Developers and Users,
I am carrying out an experiment using different search spaces (each corresponds to a specific type of architecture, e.g. U-net). Thus, does the sampler (i.e. TPE) consider trials/observations in each search space equally? In other words, can the predictive functions of TPE fed with the hyperparameter sets of search space A (and objective values) be used to search another search space B?

For example, consider 2 search spaces A and B, the objective functions for each search space are 'objective_A' and 'objective_B' respectively. If I study.optimize(objective_A, n_trials= 100), then study.optimize(objective_B, n_trials=100) still have the same TPE functions?

Thank you for your time!

**Comment by HideakiImamura** [comment, marked as answer]
Yes, it is. The different search spaces are simultaneously considered by TPE.

Note that the posterior distribution for space A and B is simply decomposed to the product of that of A and B. I mean they are independent `p(A and B) = p(A) p(B)`. However, since TPE makes a ranking of trials with the space A and B in sampling, their correlation are taken in to account for the algorithm.

**Reply by AbrightWay** [reply]
Thank you for the answer!
